libdnf/rpm/reldep_list.cpp

```
@@ -0,0 +1,139 @@
+{
+    return ReldepId(pImpl->queue[index]);
+}
+
+ReldepList & ReldepList::operator=(ReldepList && src) noexcept
+{
+    pImpl.swap(src.pImpl);
+    return *this;
+}
+
+bool ReldepList::operator!=(const ReldepList & other) const noexcept { return !(*this == other); }
+bool ReldepList::operator==(const ReldepList & other) const noexcept
+{
+    auto & this_queue = pImpl->queue;
+    auto & other_queue = other.pImpl->queue;
+    auto this_count = this_queue.size();
+    if (this_count != other_queue.size())
+        return false;
+
+    for (int i = 0; i < this_count; i++) {
+        if (this_queue[i] != other_queue[i]) {
+            return false;
+        }
+    }
+
+    return pImpl->sack->pImpl->pool == other.pImpl->sack->pImpl->pool;
+}
+
+ReldepList & ReldepList::operator=(const ReldepList & src)
+{
+    pImpl->queue = src.pImpl->queue;
+    pImpl->sack = src.pImpl->sack;
+    return *this;
+}
+
+void ReldepList::add(Reldep & reldep)
+{
+    pImpl->queue.push_back(reldep.id.id);
+}
+
+void ReldepList::add(ReldepId id)
+{
+    pImpl->queue.push_back(id.id);
+}
+
+bool ReldepList::add_reldep_with_glob(const std::string & reldep_str)
+{
+    solv::ReldepParser dep_splitter;
+    if(!dep_splitter.parse(reldep_str))
+        return false;
+    Dataiterator di;
+    Pool *pool = pImpl->sack->pImpl->pool;
+
+    dataiterator_init(&di, pool, 0, 0, 0, dep_splitter.get_name_cstr(),
+                      SEARCH_STRING | SEARCH_GLOB);
+    while (dataiterator_step(&di)) {
+        ReldepId id = Reldep::get_reldep_id(pImpl->sack, di.kv.str, dep_splitter.get_evr_cstr(),
+                                            dep_splitter.get_cmp_type());
+        add(id);
+    }
+    dataiterator_free(&di);
+    return true;
+}
+
+bool ReldepList::add_reldep(const std::string & reldep_str)
+{
+    try {
+        ReldepId id = Reldep::get_reldep_id(pImpl->sack, reldep_str);
+        add(id);
+        return true;
+    }
+    catch (...) {
```

Using libdnf specific exceptions and catch only they is better. Probably we do not want silently catch all errors eg. "memory overflow".
Add `TODO:`

---

Added TODO: